core/src/Cabana_Hammersley.hpp

```diff
@@ -0,0 +1,85 @@
+/****************************************************************************
+ * Copyright (c) 2023 by the Cabana authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+/*!
+  \file Cabana_Hammersley.hpp
+*/
+#ifndef CABANA_HAMMERSLEY_HPP
+#define CABANA_HAMMERSLEY_HPP
+
+#include <Kokkos_Vector.hpp>
+
+namespace Cabana {
+
+uint64_t KOKKOS_INLINE_FUNCTION reverseBits64(uint64_t n) {
```

Comment:
The most recent Kokkos release has bit manipulations - would be nice if we can use (probably restricted by version initially)

Reply:
https://kokkos.github.io/kokkos-core-wiki/API/core/numerics/bit-manipulation.html does not list bit reversal?